### DISCUSSION TITLE
In which folder should I place the component?

### DISCUSSION THREAD

**Discussion by sangemi** [nuxt/nuxt#16452] [Questions] [answered]
pages folder structure
- building
--- apartment
--- school
- bridge

Let's assume that there is a component "Door" that is used only in subpages of building . (Door component is not used in bridge page)

I preferred to create the door component **in the pages/building folder** rather than creating a building folder **in the components directory**.

In nuxt2, if you add "-" (dash) in front of the file name, it is not recognized as a page.

So I create Door component like /pages/building/-Door.vue path and import it in apartment.vue and school.vue.

Question 1: When I manually import the pages/building/-Door.vue component in the apartment.vue file, it works fine, but the hot loading does not work!

Question 2: Is my method mentioned above not good? Are you all placing them in the components folder?

**Comment by TheAlexLichter** [comment, marked as answer]
I'd personally place them in `components` to benefit from the auto import and have a clean separation between pages and components. But you could also use a "sub-application" via `extends` to apply e.g. DDD and get a more concerned-focused structure.

**Comment by sangemi** [comment]
I also like this method. I think it could be a little bit simpler.
https://github.com/davestewart/nuxt-areas
https://github.com/nuxt/framework/discussions/8530
